omegaconf/_utils.py

```diff
@@ -771,3 +746,76 @@ def is_generic_dict(type_: Any) -> bool:
     :return: bool
     """
     return is_dict_annotation(type_) and len(get_dict_key_value_types(type_)) > 0
+
+
+def validate_value_in_annotation(value: Any, ref_type: Any) -> bool:
+    from omegaconf import DictConfig, ListConfig
+
+    from .base import Node
+
+    valid_type = False
```

Comment:
I am not sure about it. seems like you will end up replicating the validation logic in the existing nodes/DictConfig/ListConfig.

For example, IntegerNode accepts "1" as a valid integer.
DictConfig supports both User (class) and User() as a valid input.

Reply:
Also, what about MISSING (???), what about None?
I suspect you should somehow reuse the existing validation logic here and not recreate it.

Reply:
Assignment of new values aren't wrapped nodes so I thought it should check against the type annotation.

Reply:
which would make this illegal?

```
@dataclass
class C:
  x : Union[int, float] = MISSING
```